### PR TITLE
[WFLY-17599] validate NamingBindingResourceDefinition.TYPE from allow…

### PR DESCRIPTION
**Pull request by soul2zimate**
…ed values

Issue: https://issues.redhat.com/browse/WFLY-17599

### CODE REVIEW

File: naming/src/main/java/org/jboss/as/naming/subsystem/NamingBindingResourceDefinition.java [hunk 1]
```diff
@@ -56,6 +57,8 @@
  */
 public class NamingBindingResourceDefinition extends SimpleResourceDefinition {
 
+    private static final String[] ALLOWED_TYPES = {"char", "java.lang.Character", "byte", "java.lang.Byte", "short", "java.lang.Short", "int", "java.lang.Integer", "long", "java.lang.Long", "float", "java.lang.Float", "double", "java.lang.Double", "boolean", "java.lang.Boolean", "java.lang.String", "java.net.URL"};
```
Comment (rhusar) [inline naming/src/main/java/org/jboss/as/naming/subsystem/NamingBindingResourceDefinition.java:60]: Any reason why not to use  `java.lang.Class#getName` e.g. `Character.class.getName()`, etc?

Comment (soul2zimate) [reply naming/src/main/java/org/jboss/as/naming/subsystem/NamingBindingResourceDefinition.java:60]: This list is copied from `NamingBindingAdd.coerceToType` method, but yes, we can use that approach as well.

**Review by darranl — CHANGES_REQUESTED** [review]
If this is restricting the list of allowed values based on what was previously allowed I think this will at the very least need a model bump.

As this is more restrictive than previously we likely don't need transformers for the change.

**Comment by soul2zimate** [comment]
Does it ? This only moves the check earlier to `Stage.Model` rather than `Stage.Runtime`.  An invalid value previously fails at `NamingBindingAdd.coerceToType`

**Review by bstansberry — APPROVED** [review]
I agree @soul2zimate. The Stage.RUNTIME validation logic used for this attribute is unchanged since 2012, so I can't see how doing the validation in Stage.MODEL can break any existing working use case. Technically, bumping the API version is the right thing to do, but I think it's overkill. The primary thing this does is get the read-resource-description content to reflect the long-standing actual behavior.

**Comment by bstansberry** [comment]
Thanks @soul2zimate

**Pull request merged**